server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/ElasticSearchMailboxModule.java

```diff
@@ -77,11 +81,17 @@ protected void configure() {
     @Singleton
     protected Client provideClientProvider(FileSystem fileSystem, AsyncRetryExecutor executor) throws ConfigurationException, FileNotFoundException, ExecutionException, InterruptedException {
         PropertiesConfiguration propertiesReader = new PropertiesConfiguration(fileSystem.getFile(ES_CONFIG_FILE));
+        int maxRetries = propertiesReader.getInt("elasticsearch.retryConnection.maxRetries", DEFAULT_CONNECTION_MAX_RETRIES);
+        int minDelay = propertiesReader.getInt("elasticsearch.retryConnection.minDelay", DEFAULT_CONNECTION_MIN_DELAY);
 
-        ClientProvider clientProvider = connectToCluster(propertiesReader);
+        return RetryExecutorUtil.retryOnExceptions(executor, maxRetries, minDelay, NoNodeAvailableException.class)
+            .getWithRetry(ctx -> connectToCluster(propertiesReader))
+            .get();
+    }
 
-        Client client = getRetryer(executor, propertiesReader)
-                .getWithRetry(ctx -> clientProvider.get()).get();
+    private Client createClientAndIndex(ClientProvider clientProvider, PropertiesConfiguration propertiesReader) {
```

Comment:
Hum you introduce a new private method `createClientAndIndex`, that is never used. Is it a mistake ?